src/pathfinder/water_regions.cpp

```diff
@@ -38,6 +38,10 @@ static inline int GetWaterRegionMapSizeY() { return Map::SizeY() / WATER_REGION_
 static inline TWaterRegionIndex GetWaterRegionIndex(int region_x, int region_y) { return GetWaterRegionMapSizeX() * region_y + region_x; }
 static inline TWaterRegionIndex GetWaterRegionIndex(TileIndex tile) { return GetWaterRegionIndex(GetWaterRegionX(tile), GetWaterRegionY(tile)); }
 
+using TWaterRegionPatchLabelArray = std::array<TWaterRegionPatchLabel, WATER_REGION_NUMBER_OF_TILES>;
+
+static std::unique_ptr<TWaterRegionPatchLabelArray> _spare_labels;
```

Comment:
Can be made a static local variable of ForceUpdate, for a little bit of extra encapsulation

Reply:
What is the purpose of `_spare_labels`? Seems to be to avoid a memory deallocation/allocation, rather than actually necessary.

Is the added complexity measurably necessary? A comment about its purpose would not go amiss.

Reply:
It's not going to make any enormous difference on any benchmark. I'm happy to just remove it for simplicity's sake.